networking/src/p2p/peer.rs

```diff
@@ -206,6 +208,192 @@ struct Network {
     socket_address: SocketAddr,
 }
 
+const THROTTLING_QUOTA_NUM: usize = 18;
+
+const THROTTLING_QUOTA_STRS: [&str; THROTTLING_QUOTA_NUM] = [
+    "Disconnect",
+    "Advertise",
+    "SwapRequest",
+    "SwapAck",
+    "Bootstrap",
+    "GetCurrentBranch",
+    "CurrentBranch",
+    "Deactivate",
+    "GetCurrentHead",
+    "CurrentHead",
+    "GetBlockHeaders",
+    "BlockHeader",
+    "GetOperations",
+    "Operation",
+    "GetProtocols",
+    "Protocol",
+    "GetOperationsForBlocks",
+    "OperationsForBlocks",
+];
+
+const THROTTLING_QUOTA_RESET_MS_DEFAULT: u64 = 5000; // 5 secs
+
+lazy_static::lazy_static! {
+    static ref THROTTLING_QUOTA_DISABLE: bool = {
+        match std::env::var("THROTTLING_QUOTA_DISABLE") {
+            Ok(v) => v.parse::<bool>().unwrap_or(false),
+            _ => false,
+        }
+    };
+
+    /// Quota reset period, in ms
+    static ref THROTTLING_QUOTA_RESET_MS: u64 = {
+        match std::env::var("THROTTLING_QUOTA_RESET_MS") {
+            Ok(v) => v.parse().unwrap_or(THROTTLING_QUOTA_RESET_MS_DEFAULT),
+            _ => THROTTLING_QUOTA_RESET_MS_DEFAULT,
+        }
+    };
+
+    /// Quota for tx/rx messages per [THROTTLING_QUOTA_RESET_MS]
+    static ref THROTTLING_QUOTA_MAX: [(isize, isize); THROTTLING_QUOTA_NUM] = {
+        let mut default = [
+            (1, 1), // Disconnect
+            (1, 1), // Advertise
+            (10, 10), // SwapRequest
+            (10, 10), // SwapAck
+            (1, 1), // Bootstrap
+            (10, 500), // GetCurrentBranch
+            (200, 10), // CurrentBranch
+            (10, 10), // Deactivate
+            (10, 10), // GetCurrentHead
+            (500, 500), // CurrentHead
+            (5000, 5000), // GetBlockHeaders
+            (5000, 5000), // BlockHeader
+            (500, 5000), // GetOperations
+            (20000, 10000), // Operation
+            (10, 10), // GetProtocols
+            (10, 10), // Protocol
+            (5000, 5000), // GetOperationsForBlocks
+            (10000, 10000), // OperationsForBlocks
+        ];
+        for (i, s) in THROTTLING_QUOTA_STRS.iter().enumerate() {
+            let var = "THROTTLING_QUOTA_".to_owned() + &s.to_uppercase();
+            if let Ok(val) = std::env::var(var).or_else(|_| std::env::var("THROTTLING_QUOTA_MAX")) {
+                let q = val.split(",").collect::<Vec<_>>();
+                if q.len() == 2 {
+                    if let (Ok(tx), Ok(rx)) = (q[0].parse::<isize>(), q[1].parse::<isize>()) {
+                        default[i] = (tx, rx);
+                    }
+                }
+            }
+        }
+        default
+    };
+}
+
+struct ThrottleQuota {
```

Comment:
@akoptelov 
I would move this all throttling stuff, to separate file:
`quota.rs` or `throttle_quota.rs`